src/Command/CommandHelper.php

```diff
@@ -210,6 +212,21 @@ public static function bindConfigureOptionsFromPackage(Command $command, Package
         self::validateInput($input, $command);
     }
 
+    public static function ensureOsFamilyCompatibilityFromPackage(Package $package): void
```

Comment:
This should be done in `\Php\Pie\DependencyResolver\ResolveDependencyWithComposer` - example in `\Php\Pie\DependencyResolver\ResolveDependencyWithComposer::assertCompatibleThreadSafetyMode`